pkg/metrics/tags.go

```diff
@@ -48,45 +48,93 @@ func valueOrUnknown(v string) string {
 	return metricskey.ValueUnknown
 }
 
+type revisionCtx struct {
```

Comment:
Why not just NamespacedName? Same for podctx?
What is the theoretical possibility for an activator pod and knative revision to match?